sympy/core/mul.py

```diff
@@ -1426,9 +1426,27 @@ def _eval_is_polar(self):
             all(arg.is_polar or arg.is_positive for arg in self.args)
 
     def _eval_is_extended_real(self):
+        from sympy.simplify.radsimp import fraction
+        n, d = fraction(self)
```

Comment:
It is problematic to have anything that leads to evaluation during assumptions queries. Maybe this should use `exact=True`.

It's also problematic to have circular imports like this. Maybe the `fraction` function should be defined her in `mul.py` rather than `radsimp.py`.